src/python/WMCore/Storage/StageInMgr.py

```diff
@@ -204,91 +210,56 @@ def __call__(self, **fileToStage):
         msg = "Unable to stage out file:\n"
         msg += fileToStage['LFN']
         raise StageOutFailure(msg, **fileToStage)
-
-
-    def localStageIn(self, lfn, override = None):
+    
+    def stageIn(self, lfn, stageOut_rfc=None):
         """
-        _localStageOut_
-
-        Given the lfn and local stage out params, invoke the local stage in
-        i.e. stage in lfn to pfn
-
-        if override is used the follwoing params should be defined:
+        Given the lfn and a pair of stage out and corresponding Rucio file catalog, stageOut_rfc, or override configuration invoke the stage in
+        If use override configuration self.overrideConf should contain:
         command - the stage out impl plugin name to be used
         option - the option values to be passed to that command (None is allowed)
         lfn-prefix - the LFN prefix to generate the PFN
         phedex-node - the Name of the PNN to which the file is being xferred
-
+        :param lfn: logical file name
+        :param stageOut_rfc: a pair of stage out attributes and corresponding Rucio file catalog
         """
         localPfn = os.path.join(os.getcwd(), os.path.basename(lfn))
-
-        if override:
-            pnn = override['phedex-node']
-            command = override['command']
-            options = override['option']
-            pfn = "%s%s" % (override['lfn-prefix'], lfn)
+        if self.override:
+            pnn = self.overrideConf['phedex-node']
+            command = self.overrideConf['command']
+            options = self.overrideConf['option']
+            if self.overrideConf['lfn-prefix'] is None:
+                msg = "Unable to match lfn to pfn during stage in because override lfn-prefix is None: \n %s" % lfn
+                raise StageOutFailure(msg, LFN=lfn)
+            pfn = "%s%s" % (self.overrideConf['lfn-prefix'], lfn)
             protocol = command
         else:
-            pnn = self.siteCfg.localStageOut['phedex-node']
-            command = self.siteCfg.localStageOut['command']
-            options = self.siteCfg.localStageOut.get('option', None)
-            pfn = self.searchTFC(lfn)
-            protocol = self.tfc.preferredProtocol
+            if stageOut_rfc is None:
```

Comment:
Perhaps we could change this line to `if not stageOut_rfc:`, such that it can capture a potential empty list as well?

Reply:
Done